### DISCUSSION TITLE
Setting PDF fonts to the same size as Windows fonts

### DISCUSSION THREAD

**Discussion by jwes44** [pymupdf/PyMuPDF#2972] [Looking for help]
I'm trying to hack a script to write to those PDFs that look like forms, but aren't. Since I can't find a way to edit PDF text, I write each page to a pixmap, use tkinter to display text on it, and when it's right, use insert_text to write it to the pdf. My problem is to ensure that the PDF fonts are the same size as Windows fonts. It turns out that fontsize is not points as I would have expected, or exactly pixels, e.g. to get the same size as Windows Arial 11, I have to set the font size in insert_text to 15.215 after using insert_font to load it. Is there a formula for the equivalent size in a PDF to a Windows font?

**Comment by JorjMcKie** [comment]
There is a `fitz.Rect.torect()` method which returns a matrix like this: `mat = src_rect.torect(tar_rect)`.
You can use it to convert coordinates of any rectangle or point from a source to some target coordinate system.
For example `mat = pix.irect.torect(page.rect)` converts pixmap coordinates to page coordinates, i.e. any bbox on the image corresponds to the bbox `page_bbox = pix_bbox * mat` on the page.
The quotient `page.rect.height / pixmap.irect.height` should deliver the correct value to multiply the pixmap fontsize with.

Have you considered to use redaction annotations for text replacement? If I not missing something crucial in your description, this should be **_much_** easier ...

**Reply by jwes44** [reply]
The formula turns out to be (tkfont.ascender+tkfont.descender)/(pdffont.ascender-pdffont.descender). There is a small problem with character spacing, where it is slightly larger in PDF. Is there a way to change T<sub>c</sub> for insert_text?

**Reply by JorjMcKie** [reply]
No, that is not supported.
You can play around with a scaling matrix in the morph parameter, eg `page.insert_text(pos, text, morph=(pos, matrix))`.
And set `matrix = fitz.Matrix(t, 1)`. If t is just a small bit larger or smaller than 1, the characters are a bit wider or slimmer - without changing the height.
